src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -542,8 +529,14 @@ protected void retrieve(@CheckForNull SCMSourceCriteria criteria, @NonNull SCMHe
                 listener.getLogger().format("Connecting to %s using %s%n", getServerUrl(),
                         CredentialsNameProvider.name(scanCredentials));
             }
-            // this has the side effect of ensuring that repository type is always populated.
-            listener.getLogger().format("Repository type: %s%n", WordUtils.capitalizeFully(getRepositoryType().name()));
+
+            if(cloneLinks == null) {
+                BitbucketRepository r = buildBitbucketClient().getRepository();
+                Map<String, List<BitbucketHref>> links = r.getLinks();
+                if (links != null && links.containsKey("clone")) {
+                    cloneLinks = links.get("clone");
+                }
+            }
```

Comment:
Likewise, I don't think this is necessary. In previous versions, if the plugin made an HTTP request to get the repository type and got the clone links for free in the same response, it made sense to save them and not have to make a separate request later.

Reply:
Removing this will lead in changed logic. If some error during link fetching is occured, then exception will be thrown and the code below will not be executed. Sadly, i've no idea, if this was implemented on purpose or not. I've no strong preference about leave or remove.